auraed/src/cells/cell_service/cells/cell.rs

```diff
@@ -229,6 +233,11 @@ impl CellsCache for Cell {
 
         children.broadcast_kill()
     }
+
+    fn entries(&self) -> Vec<&Cell> {
+        //self.cache.iter().collect_vec()
+        todo!()
```

Comment:
No need to actually do work here. Follow the pattern you see in the other functions. Check if we are in the allocated state, if so, call the function on children.

Reply:
Cool, I was planning to use that same pattern. I was not sure how to test this, though. Maybe I'm not fully understanding the nested `Cell` hierarchy yet. Do we have a way of creating cells within cells?

Reply:
Yes, a cell will be nested or not based on its name. So "my-cell" is a top level cell, and "my-cell/child-cell" is a cell nested in "my-cell".

At the top, `CellService` has a `CellsCache` as implemented by the `Cells` struct. The `Cells` struct has an `Option<CellName>` of its parent `Cell` (`None` for the top `Cells`). When "my-cell" is allocated, it does not have a parent, and neither does the top `CellsCache`, so the allocation (creating the cell) is handled right away. When "my-cell/child-cell" is allocated, the top `CellsCache` sees that the parent names differ, so it proxies to the appropriate immediate child cell (this proxying would continue multiple times if we had, for example, "grandparent/parent/child"; the immediate child of "grandparent" being "parent"). Need to finish this comment later, I'm losing power

Reply:
Thanks, this is super useful! Will add a test case for the nested cells to drive this implementation.

Reply:
Just double checking here, I didn't end up using `proxy_if_needed` macro I realized as it looked like the `CellsCache` contained the entire hierarchy.

Reply:
It does. Your approach was surprising to me, and just eyeballing it, I was doubting if it works (which is why I tried to test it with more deeply nested cells and led me to find the bug that I had in CellName::to_child). If your approach works, then I don't think you need proxy_if_needed.

Reply:
Added a test for a triple nested cell and that seems to work just fine.